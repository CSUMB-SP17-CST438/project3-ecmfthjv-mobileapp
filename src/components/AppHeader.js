import React from 'react';
import { View } from 'react-native';
import { Header, Title, Button, Left, Right, Body, Icon } from 'native-base';

const AppHeader = () => {
  return (
    <Header>
    </Header>
  );
};


export default AppHeader;
