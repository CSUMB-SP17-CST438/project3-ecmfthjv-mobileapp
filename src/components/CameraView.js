'use strict';
import React, { Component } from 'react';
import { Dimensions, StyleSheet, Text, TouchableHighlight, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import ActionButton from 'react-native-action-button';
import { Container, Content, Toast } from 'native-base';
import Camera from 'react-native-camera';

class BasicCamera extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Camera
          ref={(cam) => {
            this.camera = cam;
          }}
          style={styles.preview}
          aspect={Camera.constants.Aspect.fill}>
          <Text style={styles.capture} onPress={this.takePicture.bind(this)}>[CAPTURE]</Text>
        </Camera>
      </View>
    );
  }

  takePicture() {
    this.camera.capture()
      .then((data) => console.log(data))
      .catch(err => console.error(err));
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
  },
  preview: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center'
  },
  capture: {
    flex: 0,
    backgroundColor: '#fff',
    borderRadius: 5,
    color: '#000',
    padding: 10,
    margin: 40
  }
});

export default class CameraView extends Component {
    render() {
        return (
            <Container>
                <BasicCamera />
                <ActionButton
                    buttonColor="rgba(231,76,60,1)"
                    onPress={()=> Toast.show({
                        text: 'Whoop!',
                        position: 'bottom'
                    });
                }>
            </Container>
        );
    }
}
