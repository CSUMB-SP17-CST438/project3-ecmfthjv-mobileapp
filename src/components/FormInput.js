import React, { Component } from 'react';
import { View } from 'react-native';
import { Input, Label, Item } from 'native-base';

const FormInput = ({ labelProp, value, onChangeText }) => {
  return (
        <Item floatingLabel last>
            <Label>{ labelProp }</Label>
            <Input
                value={value}
                onChangeText={onChangeText}
            />
        </Item>
  );
};

export { FormInput };
