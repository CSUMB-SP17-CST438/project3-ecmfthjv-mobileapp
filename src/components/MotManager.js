import React, { Component } from 'react';
import { Text,View} from 'react-native';
import { accelerationObservable } from '../lib/sensors';
var gyroscopeObservable = require('../lib/sensors').gyroscopeObservable();

export default class MotManager extends Component {
  constructor(props) {
    super(props);
    this.state = {
      acceleration: {
        x: 'unknown',
        y: 'unknown',
        z: 'unknown',
      },
      gyroscope: {
        x: 'unknown',
        y: 'unknown',
        z: 'unknown',
      }
    };
  }

  componentWillMount() {
    accelerationObservable
      .subscribe(acceleration => this.setState({
        acceleration,
      }));

    if(gyroscopeObservable) {
      gyroscopeObservable
        .subscribe(gyroscope => this.setState({
          gyroscope,
        }));
    }
  }

  render() {
    const {
      acceleration,
      gyroscope,
    } = this.state;

    return (
      <View>
        <Text>
          Acceleration:
        </Text>
        <Text>
          {acceleration.x + '/' + acceleration.y + '/' + acceleration.z}
        </Text>
        <Text>
          Gyroscope:
        </Text>
        <Text>
          {gyroscopeObservable ? gyroscope.x + '/' + gyroscope.y + '/' + gyroscope.z : 'Gyroscope not found!'}
        </Text>
      </View>
    );
  }

  componentWillUnmount() {
    accelerationObservable.stop();
    if(gyroscopeObservable){
      gyroscopeObservable.stop();
    }
  }
}
