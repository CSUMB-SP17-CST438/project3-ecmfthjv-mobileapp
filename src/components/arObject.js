import React, { PropTypes, Component } from 'react';
import { StyleSheet, Image, View } from 'react-native';
import { mixins, variables } from '../constants';

export default class ArObject extends Component {

  constructor(props){
    super(props);
    this.state = {
      offScreenLeft: false,
      offScreenRight: false
    }
  }
  componentWillReceiveProps(nextProps) {
    if((nextProps.startingPosX + nextProps.xOffset) < 0) {
      this.setState({ offScreenLeft: true });
    } else {
      this.setState({ offScreenLeft: false });
    }
    if((nextProps.startingPosX + nextProps.xOffset) > variables.SCREEN_WIDTH) {
      this.setState({ offScreenRight: true });
    } else {
      this.setState({ offScreenRight: false });
    }
  }
  shouldComponentUpdate(nextProps) {
    return (
      this.props.xOffset != nextProps.xOffset ||
      this.props.yOffset != nextProps.yOffset
    )
  }

  render() {
    return (
      <View style={styles.root}>
        <Image
          source={require('../assests/images/marker.png')}
          resizeMode='contain'
          style={[
            styles.arTarget,
            {
              top: this.props.startingPosY + this.props.yOffset,
              left: this.props.startingPosX + this.props.xOffset
            }
          ]}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  root: {
    ...mixins.arObject,
    ...mixins.row
  },
  arTarget: {
    ...mixins.arObject,
    height: variables.AR_OBJECT_SIZE,
    width: variables.AR_OBJECT_SIZE
  }
});
