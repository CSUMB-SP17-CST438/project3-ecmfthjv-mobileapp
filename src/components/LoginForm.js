import React, { Component } from 'react';
import { View } from 'react-native';
import { Container, Content, Form, Item, Input, Label, Button, Text, Card } from 'native-base';
import { FormInput } from './FormInput';
import firebase from 'firebase';

class LoginForm extends Component {
  state = {
    email: '',
    password: ''
  };

  onButtonPress() {
    const { email, password } = this.state;
      firebase.auth().signInWithEmailAndPassword(email, password);
  }

  render() {
        return (
                <Content>
                  <Form>
                  <Card>
                      <FormInput
                          labelProp='email'
                          value={this.state.email}
                          onChangeText={ email => this.setState({ email }) }
                      />
                  </Card>
                  <Card>
                      <FormInput
                          labelProp='password'
                          value={this.state.password}
                          onChangeText={ password => this.setState({ password }) }
                      />
                  </Card>
                        <Card>
                            <Button full success onPress={this.onButtonPress.bind(this)}>
                                <Text>Log In</Text>
                            </Button>
                        </Card>
                    </Form>
                </Content>
        );
    }
}
export default LoginForm;
