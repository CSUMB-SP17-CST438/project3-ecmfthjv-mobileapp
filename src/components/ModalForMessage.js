import React from 'react';
import Modal from 'react-native-simple-modal';
import {Text, TouchableOpacity, View} from 'react-native';
import { Content, Form, Button, Text, Card } from 'native-base';
import Icon from 'react-native-vector-icons/Ionicons';
import ActionButton from 'react-native-action-button';
import { FormInput } from './FormInput';

class ModalForMessage extends React.Component {
  state = {
    open: false,
    message: ''
  };
  render() {
    return (
    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
      <ActionButton
          buttonColor="rgba(231,76,60,1)"
          onPress={() => this.setState({open: true})}>
        <Icon name="star-half" style={styles.actionButtonIcon} />
      </ActionButton>
      <Modal
        offset={this.state.offset}
        open={this.state.open}
        modalDidOpen={() => console.log('modal did open')}
        modalDidClose={() => this.setState({open: false})}
        style={{alignItems: 'center'}}>
        <View>
          <Text style={{fontSize: 20, marginBottom: 10}}>ModalForMessage Component</Text>
          <Form>
            <Card>
                <FormInput
                    labelProp='enter a message'
                    value={this.state.message}
                    onChangeText={ email => this.setState({ message }) }
                />
            </Card>
            <Card>
                <Button full success onPress={() => this.setState({open: false})}>
                    <Text>Post</Text>
                </Button>
            </Card>
          </Form>
        </View>
      </Modal>
    </View>
    );
  }
}

const styles = StyleSheet.create({
  actionButtonIcon: {
    flex:1,
    fontSize: 20,
    height: 22,
    color: 'white',
  }
});

export default ModalForMessage;
