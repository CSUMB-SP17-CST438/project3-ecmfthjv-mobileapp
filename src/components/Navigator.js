import React, { Component } from 'react';
import { Footer, FooterTab, Button, Icon, Text, Container, Content } from 'native-base';
import LoginForm from './LoginForm';
import SettingsView from './SettingsView';
import CameraView from './CameraView';

export default class Navigator extends Component {
    constructor(props){
        super(props);
        this.state = {
            tab1: false,
            tab2: true,
            tab3: false,
            ComponentPicked: LoginForm
        };
    }
    toggleTab1(){
        this.setState({
            tab1: true,
            tab2: false,
            tab3: false,
            ComponentPicked: CameraView
        });
    }
    toggleTab2(){
        this.setState({
            tab1: false,
            tab2: true,
            tab3: false,
            ComponentPicked: LoginForm
        });
    }
    toggleTab3(){
        this.setState({
            tab1: false,
            tab2: false,
            tab3: true,
            ComponentPicked: SettingsView
        });
    }
    render() {
        return (
            <Container>
                <Content>
                    <this.state.ComponentPicked/>
                </Content>
                <Footer>
                <FooterTab>
                    <Button active={this.state.tab1} onPress={() => this.toggleTab1()}>
                        <Icon active={this.state.tab1} name="camera" />
                        <Text>Camera</Text>
                    </Button>
                    <Button active={this.state.tab2} onPress={() => this.toggleTab2()} >
                        <Icon active={this.state.tab2} name="ios-person" />
                        <Text>Logging</Text>
                    </Button>
                    <Button active={this.state.tab3} onPress={() => this.toggleTab3()} >
                        <Icon active={this.state.tab3} name="settings" />
                        <Text>Settings</Text>
                    </Button>
                </FooterTab>
            </Footer>
        </Container>
        );
    }
}
