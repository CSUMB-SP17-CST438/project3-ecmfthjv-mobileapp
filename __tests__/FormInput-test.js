import 'react-native';
import React from 'react';
import { FormInput } from '../src/components/FormInput';
import renderer from 'react-test-renderer';

test('FormInput component renders correctly', () => {
  const tree = renderer.create(
    <FormInput />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
