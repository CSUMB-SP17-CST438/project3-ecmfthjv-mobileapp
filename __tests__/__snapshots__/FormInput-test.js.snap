// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "#D9D5DC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginLeft": 2,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "left": 15,
        "opacity": 1,
        "paddingBottom": undefined,
        "paddingTop": undefined,
        "position": "absolute",
        "right": 0,
        "top": 18,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#575757",
          "fontSize": 17,
          "paddingRight": 5,
          "top": 8,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 17,
          }
        }
        virtual={undefined}
      />
    </Text>
  </View>
  <TextInput
    autoCorrect={false}
    editable={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholderTextColor="#575757"
    secureTextEntry={undefined}
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "fontSize": 17,
        "height": 60,
        "lineHeight": 24,
        "top": 8,
      }
    }
    underlineColorAndroid="rgba(0,0,0,0)"
    value={undefined}
    virtual={undefined}
  />
</View>
`;
